dali/pipeline/operators/crop/bbox_crop.h

```diff
@@ -164,42 +181,40 @@ class RandomBBoxCrop : public Operator<Backend> {
     return std::make_pair(candidate_boxes, candidate_labels);
   }
 
-  std::tuple<Crop, BoundingBoxes, std::vector<int>> FindProspectiveCrop(
+  ProspectiveCrop FindProspectiveCrop(
       const BoundingBoxes &bounding_boxes, const std::vector<int> &labels,
-      float minimum_overlap) {
-    if (minimum_overlap > 0) {
-      for (int i = 0; i < num_attempts_; ++i) {
-        // Image is HWC
-        const auto candidate_height = SampleCandidateDimension();
-        const auto candidate_width = SampleCandidateDimension();
-
-        if (ValidAspectRatio(candidate_height, candidate_width)) {
-          const auto candidate_crop =
-              SamplePatch(candidate_height, candidate_width);
-
-          BoundingBoxes candidate_boxes;
-          std::vector<int> candidate_labels;
-
-          std::tie(candidate_boxes, candidate_labels) =
-              DiscardBoundingBoxesByCentroid(candidate_crop, bounding_boxes,
-                                             labels);
-
-          if (ValidOverlap(candidate_crop, candidate_boxes, minimum_overlap)) {
-            const auto remapped_boxes =
-                RemapBoxes(candidate_crop, candidate_boxes, candidate_height,
-                           candidate_width);
-
-            return std::make_tuple(candidate_crop, remapped_boxes,
-                                   candidate_labels);
+      std::pair<float, bool> minimum_overlap) {
+    if (!minimum_overlap.second)
+      return ProspectiveCrop(true, Crop::FromLtrb(0, 0, 1, 1), bounding_boxes, labels);
+
+    for (int i = 0; i < num_attempts_; ++i) {
+      // Image is HWC
+      const auto candidate_height = SampleCandidateDimension();
+      const auto candidate_width = SampleCandidateDimension();
+
+      if (ValidAspectRatio(candidate_height, candidate_width)) {
+        const auto candidate_crop =
+            SamplePatch(candidate_height, candidate_width);
+
+        if (ValidOverlap(candidate_crop, bounding_boxes, minimum_overlap.first)) {
+        BoundingBoxes candidate_boxes;
+        std::vector<int> candidate_labels;
+
+        std::tie(candidate_boxes, candidate_labels) =
+              DiscardBoundingBoxesByCentroid(candidate_crop, bounding_boxes, labels);
+
+          if (candidate_boxes.begin() != candidate_boxes.end()) {
+            const auto remapped_boxes = RemapBoxes(
+              candidate_crop, candidate_boxes, candidate_height, candidate_width);
+            return ProspectiveCrop(true, candidate_crop, remapped_boxes, candidate_labels);
           }
         }
       }
     }
-
-    return std::make_tuple(Crop::FromLtrb(0, 0, 1, 1), bounding_boxes, labels);
+    return ProspectiveCrop(false, Crop::FromLtrb(0, 0, 1, 1), bounding_boxes, labels);
   }
 
-  const std::vector<float> thresholds_;
+  std::vector<std::pair<float, bool>> sample_options_;
```

Comment:
how about a comment saying what the bool represents?

Reply:
Done